cmd/platform/message_export.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package main
+
+import (
+	"errors"
+
+	"context"
+
+	"time"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/spf13/cobra"
+)
+
+var messageExportCmd = &cobra.Command{
+	Use:     "export",
+	Short:   "Export data from Mattermost",
+	Long:    "Export data from Mattermost in a format suitable for import into a third-party application",
+	Example: "export --format=actiance --exportFrom=12345",
+	RunE:    messageExportCmdF,
+}
+
+func init() {
+	messageExportCmd.Flags().String("format", "actiance", "The format to export data in")
+	messageExportCmd.Flags().Int64("exportFrom", -1, "The timestamp of the earliest post to export, expressed in seconds since the unix epoch.")
+	messageExportCmd.Flags().Int("timeoutSeconds", -1, "The maximum number of seconds to wait for the job to complete before timing out.")
+}
+
+func messageExportCmdF(cmd *cobra.Command, args []string) error {
+	a, err := initDBCommandContextCobra(cmd)
+	if err != nil {
+		return err
+	}
+
+	if !*a.Config().MessageExportSettings.EnableExport {
+		CommandPrintErrorln("ERROR: The message export feature is not enabled.")
+		return nil
+	}
+
+	// for now, format is hard-coded to actiance. In time, we'll have to support other formats and inject them into job data
+	if format, err := cmd.Flags().GetString("format"); err != nil {
+		return errors.New("format flag error")
+	} else if format != "actiance" {
+		return errors.New("unsupported export format")
+	}
+
+	startTime, err := cmd.Flags().GetInt64("exportFrom")
+	if err != nil {
+		return errors.New("exportFrom flag error")
+	}
+
+	timeoutSeconds, err := cmd.Flags().GetInt("timeoutSeconds")
+	if err != nil {
+		return errors.New("timeoutSeconds error")
+	}
+
+	if messageExportI := a.MessageExport; messageExportI != nil {
+		ctx := context.Background()
+		var cancel context.CancelFunc
```

Comment:
You could move this down a line to make it clear that it's only used within the `if` block.